File: programs/mandelbrot/mandelbrot.js
const ITERATIONS = 100;
const EXPECTED = 8939;

function mandelbrot() {
  let count = 0;

  let h = 150.0;
  let y = 0.0;
  let Z = 0.0;
  let z = 0.0;
  let T = 0.0;
  let t = 0.0;
  let C = 0.0;
  let c = 0.0;
  let U = 0.0;
  let V = 0.0;
  let K = 1.5;
  let k = 1.0;

  while (y < 150) {
    y += 1;
    let x = 0.0;
    while (x < 150) {
      x += 1;
      Z = 0.0;
      z = 0.0;
      T = 0.0;
      t = 0.0;
      U = x * 2;
      U /= h;
      V = y * 2;
      V /= h;
      C = U - K;
      c = V - k;

      let i = 0;
      while (i < 50) {
        i += 1;
        if (T + t <= 4) {
          z = Z * z;
          z *= 2;
          z += c;
          Z = T - t;
          Z += C;
          T = Z * Z;
          t = z * z;
        }
      }
      if (T + t <= 4) {
        count += 1;
      }
    }
  }
  return count;
}


for (let i = 0; i < ITERATIONS; i++) {
  const result = mandelbrot();
  if (result != EXPECTED) {
    throw "Error: got " + result + " expected " + EXPECTED;
  }
}

File: programs/mandelbrot/mandelbrot.c
/* The Computer Language Shootout
   http://shootout.alioth.debian.org/

   contributed by Greg Buchholz
*/

// Modified to be more similar to the Python version.


#include<stdio.h>
#include<stdlib.h>

#define W 150
#define H 150
#define ITERATIONS 10

#define EXPECTED 8939


int mandelbrot()
{
    int bit_num = 0;
    char byte_acc = 0;
    int i, iter = 50;
    double x, y, limit = 2.0;
    double Zr, Zi, Cr, Ci, Tr, Ti;
    int count = 0;

    for (y = 0; y < H; ++y) {
        for (x = 0; x < W; ++x) {
            Zr = Zi = Tr = Ti = 0.0;
            Cr = (2.0 * x / W - 1.5);
            Ci = (2.0 * y / H - 1.0);

            for (i = 0; i < iter && (Tr + Ti <= 4); ++i) {
                Zi = 2.0 * Zr * Zi + Ci;
                Zr = Tr - Ti + Cr;
                Tr = Zr * Zr;
                Ti = Zi * Zi;
            }

            if (Tr + Ti <= 4) {
                count += 1;
            }
        }
    }
    return count;
}


int main(int argc, char **argv)
{
    for (int i = 0; i < ITERATIONS; i++) {
        int result = mandelbrot();
        printf("%d\n", result);

        // FIXME: off by one !
//        if (result != EXPECTED) {
//            printf("ERROR! result=%d expected=%d", result, EXPECTED);
//            exit(1);
//        }
    }
}
